packages/product/src/services/product-module-service.ts

```diff
@@ -231,6 +244,108 @@ export default class ProductModuleService<
     return [JSON.parse(JSON.stringify(variants)), count]
   }
 
+  @InjectTransactionManager("baseRepository_")
+  async updateVariants(
+    data: ProductTypes.UpdateProductVariantOnlyDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<ProductTypes.ProductVariantDTO[]> {
+    const variants = await this.listVariants(
+      { id: data.map(({ id }) => id) },
+      { relations: ["options"] },
+      sharedContext
+    )
+    const variantsMap = new Map(
+      variants.map((variant) => [variant.id, variant])
+    )
+
+    if (variants.length !== data.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Cannot update non-existing variants with ids: " +
+          data
+            .filter(({ id }) => !variantsMap.has(id))
+            .map(({ id }) => id)
+            .join(", ")
+      )
+    }
+
+    const optionsToUpsert: (
+      | CreateProductOptionValueDTO
+      | UpdateProductOptionValueDTO
+    )[] = []
+    const optionsToDelete: string[] = []
+
+    const toUpdate = data.map(({ id, options, ...rest }) => {
+      const variant = variantsMap.get(id)!
+
+      const toUpdate: UpdateProductVariantDTO & { product_id: string } = {
+        id,
+        product_id: variant.product_id,
+      }
+
+      if (options?.length) {
+        const optionIdToUpdateValueMap = new Map(
+          options.map(({ option, option_id, value }) => {
+            const computedOptionId = option_id ?? option.id ?? option
+            return [computedOptionId, value]
+          })
+        )
+
+        for (const existingOption of variant.options) {
+          if (!optionIdToUpdateValueMap.has(existingOption.option)) {
+            optionsToDelete.push(existingOption.id)
+          } else {
+            optionsToUpsert.push({
+              id: existingOption.id,
+              option: existingOption.option as unknown as string,
+              value: optionIdToUpdateValueMap.get(existingOption.option)!,
+            })
+            optionIdToUpdateValueMap.delete(existingOption.option)
+          }
+        }
+
+        for (const [option, value] of optionIdToUpdateValueMap.entries()) {
+          optionsToUpsert.push({ option, value, variant: id })
+        }
+      }
+
+      if (Object.keys(rest).length) {
+        for (const [key, value] of Object.entries(rest)) {
+          if (variant[key] !== value) {
+            toUpdate[key] = value
+          }
+        }
+      }
```

Comment:
```suggestion
      for (const [key, value] of Object.entries(rest)) {
        if (variant[key] !== value) {
          toUpdate[key] = value
        }
      }
```

the if seems redundant